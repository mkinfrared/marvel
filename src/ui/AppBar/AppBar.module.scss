@import "styles/colors";
@import "styles/vars";

.AppBar {
  align-items: center;
  background-color: $secondary-main;
  display: flex;
  height: $header-height;
  padding: 0 24px;
  width: 100%;
}
