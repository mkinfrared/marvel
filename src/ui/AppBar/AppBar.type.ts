export interface AppBarProps {
  children?: React.ReactNode;
  className?: string;
}
