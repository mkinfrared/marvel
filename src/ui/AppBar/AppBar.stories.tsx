import { Meta, Story } from "@storybook/react/types-6-0";
import React from "react";

import Button from "ui/Button";
import Input from "ui/Input";

import { AppBar } from "./AppBar";

export default {
  title: "UI/AppBar",
  component: AppBar
} as Meta;

const Template: Story = () => (
  <AppBar>
    <Button>I am a button</Button>
    <Input value="" onChange={() => null} />
  </AppBar>
);

const Primary = Template.bind({});

export { Primary };
