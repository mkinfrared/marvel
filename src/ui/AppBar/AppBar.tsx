import React from "react";

import classes from "utils/classes";

import css from "./AppBar.module.scss";
import { AppBarProps } from "./AppBar.type";

const AppBar = ({ children, className }: AppBarProps) => {
  return <div className={classes(css.AppBar, className)}>{children}</div>;
};

export { AppBar };

export default React.memo(AppBar);
