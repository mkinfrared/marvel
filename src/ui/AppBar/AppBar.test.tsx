import { render } from "@testing-library/react";
import React from "react";

import { AppBar } from "./AppBar";

describe("<AppBar />", () => {
  const Component = <AppBar />;

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
