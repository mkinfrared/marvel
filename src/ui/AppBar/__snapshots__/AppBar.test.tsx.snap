// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> should match the snapshot 1`] = `
<div>
  <div
    class="AppBar"
  />
</div>
`;
